File: cosmol_viewer/examples/render_protein.py
from cosmol_viewer import Protein, Scene, Viewer

mmcif_data = open("./examples/6fi1.cif", "r", encoding="utf-8").read()

prot = Protein.from_mmcif(mmcif_data).rainbow_residues()

scene = Scene()
# scene.use_black_background()
scene.set_scale(0.25)
scene.recenter(prot.get_center())
scene.add_shape_with_id("prot", prot)
scene.set_background_color("#021529")
scene.set_camera_view(azimuth=180, elevation=0, distance=32, fov=18)

scene.save_image("render_protein.png", width=1200, height=900)

viewer = Viewer.render(scene, width=800, height=500)

print("Press Any Key to exit...", end="", flush=True)
_ = input()

File: cosmol_viewer/examples/render_protein.rs
use cosmol_viewer::{RenderQuality, Scene, Viewer, shapes::Protein};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let prot = Protein::from_mmcif(include_str!("../examples/6fi1.cif"))?.rainbow_residues();

    let mut scene = Scene::new();
    scene.set_scale(0.2);
    scene.recenter(prot.get_center());
    scene.add_shape_with_id("prot", prot);
    scene.set_background_color("#021529");

    // Viewer::render(&scene, 800.0, 500.0)?;
    Viewer::render_with_quality(&scene, 800.0, 500.0, RenderQuality::High)?;

    println!("Press Enter to exit...");
    use std::io::{self, Write};
    let _ = io::stdout().flush();
    let _ = io::stdin().read_line(&mut String::new());

    Ok(())
}
